test/Microsoft.AspNet.Mvc.IntegrationTests/ValidationIntegrationTests.cs

```diff
@@ -1086,5 +1162,13 @@ private static void AssertRequiredError(string key, ModelError error)
                 string.Format("The {0} field is required.", key)), error.ErrorMessage);
             Assert.Null(error.Exception);
         }
+
+        private class User
```

Comment:
Everywhere else in this test class, the target types are defined just before their first use. Not good practise in product code but it makes this long file easier to use. So move this above `Validation_FormatException_ShowsInvalidValueMessage_OnSimpleTypeProperty()`.